src/Microsoft.AspNet.Mvc.Xml/XmlSerializerOutputFormatter.cs

```diff
@@ -159,11 +166,26 @@ public override Task WriteResponseBodyAsync([NotNull] OutputFormatterContext con
                     obj = wrapperProvider.Wrap(obj);
                 }
 
-                var xmlSerializer = CreateSerializer(wrappingType);
+                var xmlSerializer = GetCachedSerializer(wrappingType);
                 xmlSerializer.Serialize(xmlWriter, obj);
             }
 
             return Task.FromResult(true);
         }
+
+        private XmlSerializer GetCachedSerializer(Type type)
```

Comment:
all these `GetCachedSerializer` need to be protected. They are breaking a creation of a different serializer per request (which the user can do). So if the user wants to create serializer per request he can override caching (or cache anyway he wants to). This also resolves @kichalla question about statics above :watch: